libultraship/libultraship/Archive.cpp

```diff
@@ -139,13 +144,16 @@ namespace Ship {
 
 	bool Archive::AddFile(const std::string& path, uintptr_t fileData, DWORD dwFileSize) {
 		HANDLE hFile;
-
+#ifdef _MSC_VER
 		SYSTEMTIME sysTime;
 		GetSystemTime(&sysTime);
 		FILETIME t;
 		SystemTimeToFileTime(&sysTime, &t);
 		ULONGLONG stupidHack = static_cast<uint64_t>(t.dwHighDateTime) << (sizeof(t.dwHighDateTime) * 8) | t.dwLowDateTime;
-
+#else
```

Comment:
wish there was a comment here giving more details about what the goal of stupidHack is. Realize it's not your code, but if you understand what's happening I'd be interested to know.

Reply:
I actually don't know what this is far. This is also very old code so I'm not sure it's even required anymore (?)

Reply:
For anyone wondering: it's packing the 32-bit high and low order portions of the Windows-style timestamp (100 nanos since January 1, 1601 UTC) into a 64-bit value.

The code does mark 0 as being a valid value if the time isn't there, but it shouldn't hurt to have it.